src/Avalonia.Controls/DefinitionBase.cs

```diff
@@ -110,11 +110,11 @@ internal void SetMinSize(double minSize)
         /// </remarks>
         internal void OnUserSizePropertyChanged(AvaloniaPropertyChangedEventArgs e)
         {
-            if (InParentLogicalTree)
+            if (Parent != null)
             {
                 if (_sharedState != null)
                 {
-                    _sharedState.Invalidate();
+                    _sharedState?.Invalidate();
```

Comment:
Isn't the '?' null check irrelevant since it already checked in the if statement?

Reply:
Yes, already fixed that in my latest commit but i havent pushed it yet